src/library/scala/concurrent/Future.scala

```diff
@@ -558,6 +577,28 @@ object Future {
     classOf[Unit]    -> classOf[scala.runtime.BoxedUnit]
   )
 
+  private[concurrent] final def coerce[F[_],A,B](f: F[A]): F[B] = f.asInstanceOf[F[B]]
+
+  private[concurrent] final val collectFailed =
+    (t: Any) => throw new NoSuchElementException("Future.collect partial function is not defined at: " + t) with NoStackTrace
+
+  private[concurrent] final val filterFailure =
+    Failure[Nothing](new NoSuchElementException("Future.filter predicate is not satisfied") with NoStackTrace)
+
+  private[concurrent] final val failedFailure =
+    Failure[Nothing](new NoSuchElementException("Future.failed not completed with a throwable.") with NoStackTrace)
+
+  private[concurrent] final val recoverWithFailedMarker =
+    Future.failed[Nothing](new Throwable with NoStackTrace)
+
+  private[concurrent] final val recoverWithFailed =
```

Comment:
Since we are pre-allocating and reusing a function instance here,
would it be beneficial to do the same for the 3 uses of `Predef.identity`, which I think still isn't cached?

Reply:
Great catch with Predef.identity: In fact, it used to use Predef.conforms since that instance was cached, but someone changed it to identity and I missed that.